source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/debounced-search-field/index.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+
+import { debounce } from '@lib/debounce';
+import { filterForValidAttributes, IAttributeProps } from '@lib/dom';
+
+export interface IProps {
+  onSubmit: (term: string) => void;
+  placeholder?: string;
+}
+
+interface IState {
+  searchTerm: string;
+}
+
+export default class DebouncedSearch extends React.Component<IProps & IAttributeProps, IState> {
```

Comment:
I'm slightly conflicted on this, because every component should have these. I'm a little sad the default Component types don't auto-union with these.
So, idk